src/utils/llm/manager.ts

```diff
@@ -27,17 +28,28 @@ class LLMManager implements LLMManagerInterface {
   private openaiProvider: OpenAIProvider
   private groqProvider: GroqProvider
   private anthropicProvider: AnthropicProvider
+  private ollamaProvider: OllamaAIOpenAIProvider
 
-  constructor(apiKeys: { openai?: string; groq?: string; anthropic?: string }) {
+  constructor(
+    apiKeys: { openai?: string; groq?: string; anthropic?: string },
+    ollamaBaseUrl?: string,
+  ) {
     this.openaiProvider = new OpenAIProvider(apiKeys.openai ?? '')
     this.groqProvider = new GroqProvider(apiKeys.groq ?? '')
     this.anthropicProvider = new AnthropicProvider(apiKeys.anthropic ?? '')
+    this.ollamaProvider = new OllamaAIOpenAIProvider(
+      apiKeys.openai ?? 'empty',
+      ollamaBaseUrl ?? '',
+    )
```

Comment:
Could you explain why we're using the OpenAI API key with Ollama? As far as I understand, Ollama doesn't require an OpenAI API key to function.

Reply:
that's correct. During some research I saw a use case of hosted ollama with auth tokens but checking - it isn't part of the core functionality of ollama.  I will rework.

Reply:
I see. Maybe we can add auth tokens later if needed. Thanks for your work.

Reply:
refactored this away in comment https://github.com/glowingjade/obsidian-smart-composer/pull/45/commits/043030b6acd437f53ac0ed5298ea6329c36c3b4a

Reply:
Nice refactoring!